system/security/shared/caching.cpp

```diff
@@ -585,6 +562,46 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
         return true;
     }
 
+    // If the cache is not ready, then an initial fill is required. Acquire the cache RW lock and make sure no other thread has
+    // tried to fill the cache. Once filled, start the fill thread and wait for it to be ready.
+    if (!m_fileScopeCacheReady)
+    {
+        WriteLockBlock writeLock(m_scopesRWLock);
+
+        if (!m_fileScopeCacheReady)
+        {
+            IArrayOf<ISecResource> scopes;
+            aindex_t count = m_secMgr->getManagedScopeTree(RT_FILE_SCOPE, nullptr, scopes);
+            removeAllManagedFileScopes();
+            if (count)
+                addManagedFileScopes(scopes);
+            time_t now;
+            time(&now);
+            m_lastCacheFillTime = now;
+
+            m_fileScopeCacheFillThread = std::thread(&CPermissionsCache::managedFileScopesCacheFillThread, this);
+
+            constexpr unsigned failTimeoutSeconds = 60;
+            constexpr unsigned threadStartWaitTimeSeconds = 5;
+            CCycleTimer timer;
+            while (true)
```

Comment:
This loop is now unnecessary- the cache has already been initialized by the code above.

Reply:
agree, and the signalling mechanism for ready isn't necessary either with this recent change.
m_fileScopeCacheReady is guaranteeing it is ready..